examples/batched/batched.cpp

```diff
@@ -155,16 +155,16 @@ int main(int argc, char ** argv) {
 
     // remember the batch index of the last token for each parallel sequence
     // we need this to determine which logits to sample from
-    std::vector<int32_t> i_batch(n_parallel, batch.n_tokens - 1);
+    std::vector<int32_t> i_batch(n_parallel, llama_batch_ext_get_n_tokens(batch) - 1);
```

Comment:
I think the next step along this refactoring is to remove all usages of `i_batch` from the examples. The `i_batch` is the index we use to extract logits from the `i`th token in the batch, but this pattern is quite cumbersome and not very intuitive. In order to avoid this pattern, we have to introduce a new API call for sampling a token from a sequence:

```c
    LLAMA_API llama_token llama_sampler_sample_seq(struct llama_sampler * smpl, struct llama_context * ctx, llama_seq_id seq_id);
```

This should be enough to replace most or all usages of `i_batch`. We can do this in a next PR.